[Zhang Yuan] iP

volleyballkickedme opened the pull request:
## FRIDAY Overview

> “The best way to predict the future is to invent it.” – Alan Kay

FRIDAY is a comprehensive task management tool designed to streamline your productivity. It’s:

- **Text-based** and easy to use
- **Highly customizable** to fit your needs
- **FAST** — saves you time 🏃‍♂️
- **User-friendly** interface

### Key Features

1. - [x] **Task Management**: Organize and track your tasks efficiently.
2. - [ ] **Deadlines Management**: Set and manage deadlines (_coming soon_).
3. - [ ] **Reminders**: Never miss an important task (_coming soon_).

To get started:

- Download it from [here](https://github.com/volleyballkickedme/ip).
- Navigate to the folder that contains the jar file and run it with  `java -jar FRIDAY.jar`.


### Sample Code

Here’s a snippet of the main method for Java programmers:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

volleyballkickedme committed:
This is level zero of the week 2 project task

volleyballkickedme committed:
This is level 1, where the bot echoes the user input

volleyballkickedme committed:
Level 2 - the bot now is able to store user input and display it in a list format on command

volleyballkickedme committed:
Level 3 allows you to mark tasks as done or undone

volleyballkickedme committed:
Level 4 provides support for 3 different kinds of tasks, todo, event and deadline

volleyballkickedme committed:
semi automated testing of the chatbot

volleyballkickedme committed:
error handling implemented

volleyballkickedme committed:
Level 6 adds the delete functionality that allows users to remove items from the list

volleyballkickedme committed:
Added input text file

volleyballkickedme committed:
Bot is able to read inputs from storage file "FRIDAY.txt"

volleyballkickedme committed:
update storage with data read from input

volleyballkickedme committed:
Implemented level 7 - reading and writing to storage

volleyballkickedme committed:
Refactored code to put repeated code in functions

volleyballkickedme committed:
Level-8 teaches the chatbot how to recognize dates for deadlines

volleyballkickedme committed:
A-MoreOOP - implemented storage class

volleyballkickedme committed:
Revert "A-MoreOOP - implemented storage class"

This reverts commit 6214f339dad8105e8b08cb803d4837d28fe44f5b.

volleyballkickedme committed:
Reapply "A-MoreOOP - implemented storage class"

This reverts commit ec121332b4ec797bb16a790c8d1129a443390f41.

volleyballkickedme committed:
A-MoreOOP created task list class

volleyballkickedme committed:
A-MoreOOP create Ui class

volleyballkickedme committed:
A-MoreOOP create Parser class

volleyballkickedme committed:
A-MoreOOP Rewrite program to be more object oriented

volleyballkickedme committed:
Add all files into package FRIDAY

volleyballkickedme committed:
Merge branch 'add-gradle-support'

volleyballkickedme committed:
Add Gradle support

volleyballkickedme committed:
Add JUnit tests

volleyballkickedme committed:
Create jar file

volleyballkickedme committed:
Add JavaDOc headers to classes and methods

volleyballkickedme committed:
Edit code to comply with coding standards

volleyballkickedme committed:
Add search functionality that allows users to search for tasks that match the keyword they input

volleyballkickedme committed:
Merge branch 'A-CodingStandard'

# Conflicts: 1
#	src/main/java/FRIDAY/Event.java

volleyballkickedme committed:
no message

volleyballkickedme committed:
Edit spelling mistake

Vangmay commented on src/main/java/FRIDAY/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+package FRIDAY;
```
I think this is a coding standard violation, as packages are only supposed to use names in lowercase.

Vangmay commented on src/main/java/FRIDAY/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+
+    @Override
+    public String toString() {
+        //print it out in MM-DD-YYYY format
```
Perhaps a javadoc comment would be suitable for this method instead.

Vangmay commented on src/main/java/FRIDAY/Storage.java:
```diff
@@ -0,0 +1,107 @@
+package FRIDAY;
+
+import java.io.*;
```
Wildcard imports are a violation of the coding standard, I suggest import every class your require individually.

Vangmay commented on src/main/java/FRIDAY/Storage.java:
```diff
@@ -0,0 +1,107 @@
+    /**
+     * method to populate the program storage with data read from the hard drive
+     *
+     * @param data string to represent all the details of the task, including its type and other miscellaneous details
```
A good idea would be to make use of `@throws` tag in javadoc to signify the `FRIDAYException` being thrown.

Vangmay commented on src/main/java/FRIDAY/Ui.java:
```diff
@@ -0,0 +1,59 @@
+import java.util.Scanner;
+
+public class Ui {
+    private static final String GREETING_MESSAGE = "Hello! I'm FRIDAY\nWhat can I do for you?\n";
```
I like the idea of storing the default messages inside constants

Vangmay commented on src/main/java/FRIDAY/FRIDAYException.java:
```diff
@@ -0,0 +1,7 @@
+package FRIDAY;
+
+public class FRIDAYException extends Exception {
```
Perhaps renaming the class to `FridayException` would be more compliant with the coding standards

Vangmay reviewed (commented):
LGTM. I think your code is really well organized and I found your code easy to read. There are a few minor coding standard violations that I have highlighted, for example naming conventions and wildcard imports.

rxchell commented on src/main/java/FRIDAY/FRIDAY.java:
```diff
@@ -0,0 +1,138 @@
+                //create new deadline task
+                Task newDeadline = Parser.parseDeadline(taskDetails);
+                taskList.addTask(newDeadline);
+                ui.printAdd(newDeadline, taskList.numTasks());
+                break;
+            case ("event"):
+                if (taskDetails.isEmpty()) {
+                    throw new FRIDAYException("ERROR: Please note that the description of a task cannot be left empty");
+                }
+                Task newEvent = Parser.parseEvent(taskDetails);
+                taskList.addTask(newEvent);
+                ui.printAdd(newEvent, taskList.numTasks());
+                break;
+            case ("delete"):
+                if (taskDetails.isEmpty()) {
+                    throw new FRIDAYException("ERROR: Please note that the description of a task cannot be left empty");
+                }
+                taskList.removeTask(Integer.parseInt(taskDetails.substring(0, 1)) - 1);
+                ui.printRemove(taskList.getTaskAt(Integer.parseInt(taskDetails.substring(0, 1)) - 1), taskList.numTasks());
+                break;
+
+            case ("bye"):
+                storage.updateStorage(this.taskList.getList());
+                deactivateBot();
+                break;
+            case ("list"):
+                System.out.println(taskList.displayTasks());
+                break;
+            case("search") :
+                String word = Parser.parseTaskDetails(taskDetails);
+                ArrayList<Task> searchResults = taskList.search(word);
+                ui.displaySearchResults(searchResults);
+            //if there is no input then nothing added to list
+            case (""):
+                break;
+            //to handle all normal inputs less empty strings
+            default:
+                throw new FRIDAYException("It appears that you have attempted to log an unrecognized class type. Please try again");
+        }
+    }
```
Is there extra spacing in the indentation of the cases?

rxchell commented on src/main/java/FRIDAY/Parser.java:
```diff
@@ -0,0 +1,49 @@
+     * @param input a String representing user input
+     * @return a String representing the type of action the proogram is meant to execute
+     */
+    public static String parseCmd(String input) {
```
Should the method name be clearer? Eg `parseCommand` instead of `parseCmd`

rxchell reviewed (commented):
Looks good. Some changes for adhering to the code style is needed.

rxchell commented on src/main/java/FRIDAY/ToDo.java:
```diff
@@ -0,0 +1,40 @@
+     * @param type the type of the task; if greater than 0, the task is complete
+     */
+    public ToDo(String description, int type) {
+        super(description, type);
+    }
+
+    /**
+     * outputs all the information pertaining the task in a specified format for storage purposes
+     * @return a string containing all important details of the task
+     */
+    public String storageDisplay() {
+        return "T" + super.storageDisplay();
+    }
+
+    @Override
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+}
```
I like how the attributes and methods are all inherited from the parent class, makes the code looks neater!

rxchell commented on src/test/java/FRIDAY/DeadlineTest.java:
```diff
@@ -0,0 +1,34 @@
+
+import java.time.LocalDate;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Is the static import supposed to be the first import before other imports?

rxchell commented on src/test/java/FRIDAY/FRIDAYTest.java:
```diff
@@ -0,0 +1,17 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class FRIDAYTest {
+    @Test
+    public void dummyTest() {
+        assertEquals(2, 2);
+    }
+
+    @Test
+    public void anotherDummyTest() {
+        assertEquals(4, 4);
+    }
+}
```
What is the dummy test for?

rxchell reviewed (commented):
Looks good to me, except for some minor styling issues.

volleyballkickedme committed:
Add checkstyle

volleyballkickedme committed:
Modify code to fit coding standards

volleyballkickedme committed:
Add skeleton for GUI

volleyballkickedme committed:
Add GUI for bot

volleyballkickedme committed:
Add GUI for bot

volleyballkickedme committed:
Add assert statements

volleyballkickedme committed:
improve code quality

volleyballkickedme committed:
Merge pull request #1 from volleyballkickedme/branch-A-Assertions

Add assert statements

volleyballkickedme committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/FRIDAY/FRIDAY.java

volleyballkickedme committed:
Merge pull request #2 from volleyballkickedme/branch-A-CodeQuality

Branch code quality

volleyballkickedme committed:
Create empty archive file locally

volleyballkickedme committed:
Add feature to archive all tasks in the current task list

volleyballkickedme committed:
Fixed abstraction issues

volleyballkickedme committed:
Update Storage.java

volleyballkickedme committed:
Update Storage.java

volleyballkickedme committed:
Update ui output

volleyballkickedme committed:
Add greeting message that teaches user how to use the bot

volleyballkickedme committed:
Add CSS files

volleyballkickedme committed:
Style bot background and text box

volleyballkickedme committed:
Update GUI

volleyballkickedme committed:
Made profile pictures circular

volleyballkickedme committed:
Polished GUI

volleyballkickedme committed:
Modify the text displayed on adding and removing tasks

volleyballkickedme committed:
Modify text displayed on checking and unchecking tasks

volleyballkickedme committed:
Update remaining text displays to give bot better personality

volleyballkickedme committed:
Remove redundant comments

volleyballkickedme committed:
Add better error handling

Several methods in the getResponse method in FRIDAY throw different errors due to incorrect format in user input.

Refactoring the code such that the individual methods throw exceptions, rather than throwing exceptions in the getResponse method improves code quality and adheres better to OOP principles.

As a step toward such refactoring, the exceptions originally thrown in the getResponse method body have all been shifted into their respective methods. All thrown exceptions are then handled when the getResponse method is called.

volleyballkickedme committed:
Fix bug that did not allow user to delete first task in list

volleyballkickedme committed:
Tweak parser to give user more accurate feedback upon invalid input

Previously program would feedback the same error message regardless of whether the command was unrecognized or if the task description was left as empty. This is not user friendly as users will be confused about what the error in their input was.

This modified code will now more accurately feedback to the user what the error in their input was so that they are better able to correct their input.

volleyballkickedme committed:
Fix bug that occured when user tries to delete a non-existent task

volleyballkickedme committed:
Add Ui screenshot

volleyballkickedme committed:
Write initial version of User Guide

volleyballkickedme committed:
Update README.md

volleyballkickedme committed:
Edit method and class headers to better comply with JavaDoc convention.

volleyballkickedme committed:
Add Ui screenshot to gitignore

volleyballkickedme committed:
Merge remote-tracking branch 'origin/master'

volleyballkickedme committed:
Update README.md to utilize more elements of GFMD

volleyballkickedme committed:
Refactor code to remove redundant variable

volleyballkickedme committed:
Update JavaDoc header

volleyballkickedme committed:
Remove redundant conditional
